### PR TITLE
Feature/3 customdomains

### LINKED ISSUE #3: Allow custom domains
### Discussed in https://github.com/Dynatrace/ace-box/discussions/2

<div type='discussions-op-text'>

<sup>Originally posted by **kristofre** June 23, 2021</sup>
At the moment, either a specific IP address or generated by Terraform is used to create a *.nip.io domain. That does not always work in corporate environments. Being able to specify a custom subdomain would allow us a way around that.</div>

### PR DESCRIPTION
**Pull request by kristofre**

### CODE REVIEW

File: terraform/gcloud/outputs.tf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 output "acebox_dashboard" {
-  value = "http://dashboard.${google_compute_instance.acebox.network_interface[0].access_config[0].nat_ip}.nip.io using dynatrace:dynatrace to log in"
+  value = (var.custom_domain != "" ? "http://dashboard.${var.custom_domain}" : "http://dashboard.${google_compute_instance.acebox.network_interface[0].access_config[0].nat_ip}.nip.io")
```
Comment (tobigremmer-dt) [inline terraform/gcloud/outputs.tf:2]: Do we want to parameterize protocol here as well, i.e. http or https?

File: terraform/azure/outputs.tf [hunk 2]
```diff
@@ -1,7 +1,11 @@
-output "acebox_azure_ip" {
+output "acebox_dashboard" {
+  value = (var.custom_domain != "" ? "http://dashboard.${var.custom_domain}" : "http://dashboard.${azurerm_public_ip.acebox_publicip.ip_address}.nip.io")
```
Comment (tobigremmer-dt) [inline terraform/azure/outputs.tf:2]: Do we want to parameterize protocol here as well, i.e. http or https?

File: terraform/aws/private_vpc/outputs.tf [hunk 3]
```diff
@@ -1,6 +1,5 @@
 output "acebox_dashboard" {
-  description = "Public IPs assigned to the EC2 instance"
-  value = "http://dashboard.${aws_instance.acebox.private_ip}.nip.io"
+  value = (var.custom_domain != "" ? "http://dashboard.${var.custom_domain}" : "http://dashboard.${aws_instance.acebox.private_ip}.nip.io")
```
Comment (tobigremmer-dt) [inline terraform/aws/private_vpc/outputs.tf:2]: Do we want to parameterize protocol here as well, i.e. http or https?

File: terraform/aws/default_vpc/outputs.tf [hunk 4]
```diff
@@ -1,6 +1,5 @@
 output "acebox_dashboard" {
-  description = "Public IPs assigned to the EC2 instance"
-  value = "http://dashboard.${aws_instance.acebox.public_ip}.nip.io"
+  value = (var.custom_domain != "" ? "http://dashboard.${var.custom_domain}" : "http://dashboard.${aws_instance.acebox.public_ip}.nip.io")
```
Comment (tobigremmer-dt) [inline terraform/aws/default_vpc/outputs.tf:2]: Do we want to parameterize protocol here as well, i.e. http or https?

File: terraform/aws/custom_vpc/outputs.tf [hunk 5]
```diff
@@ -1,6 +1,5 @@
 output "acebox_dashboard" {
-  description = "Public IPs assigned to the EC2 instance"
-  value = "http://dashboard.${aws_eip.acebox_eip.public_ip}.nip.io"
+  value = (var.custom_domain != "" ? "http://dashboard.${var.custom_domain}" : "http://dashboard.${aws_instance.acebox.public_ip}.nip.io")
```
Comment (tobigremmer-dt) [inline terraform/aws/custom_vpc/outputs.tf:2]: Do we want to parameterize protocol here as well, i.e. http or https?

File: terraform/gcloud/outputs.tf (hunk 1 above)
Comment (kristofre) [reply terraform/gcloud/outputs.tf:2]: For now https is not yet supported, so for now this would be good IMO. In the future this would be an option.

**Review by tobigremmer-dt — APPROVED** [review]
Looks good

**Comment by kristofre** [comment]
Closes https://github.com/Dynatrace/ace-box/issues/3

**Pull request merged**